File: javascript/valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    let mappy = {};
    
    if (s.length !== t.length) {
        return false;
    }
    
    for (let i = 0; i < s.length; i++) {
        if (mappy[s[i]]) {
            mappy[s[i]]++;
        } else {
            mappy[s[i]] = 1;
        }
    }
    
    for (let i = 0;  i < t.length; i++) {
        if (mappy[t[i]]) {
            mappy[t[i]]--;
        } else {
            return false;
        }
    }
    return true; 
};

File: python/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        # for them to be anagrams they need to have
        # the same # of letters, and the same character count
         #if they are not the same length, return False immediately
            
        if len(s) != len(t):
            return False
        
        sMap = {}
        tMap = {}
        
        for i in range(len(s)):
            charS, charT = s[i], t[i]
            sMap[charS] = 1 + sMap.get(charS, 0)
            tMap[charT] = 1 + tMap.get(charT, 0)
        
        if sMap == tMap:
            return True
        else:
            return False
